argus/tests/cloud/smoke.py

```diff
@@ -75,6 +80,91 @@ def test_password_set(self):
+        return stdout
+
+    def _random_password(self, length=10):
+        random_generator = random.SystemRandom()
+        alphabet = string.ascii_letters + string.digits
+        return ''.join(random_generator.choice(alphabet)
+                       for _ in range(length))
+
+    def _update_password(self, password):
+        url = "http://%(host)s:%(port)s/password" % {
+            "host": self.manager.services[1].host,
+            "port": self.manager.services[1].port
+        }
+        try:
+            params = urllib.parse.urlencode({'password': password})
+            request = urllib.request.Request(url, data=params)
+            response = urllib.request.urlopen(request)
+        except urllib.error.HTTPError as exc:
+            return exc.code
+        return response.getcode()
+
+    def _wait_for_service_status(self, status, retry_count=3,
+                                 retry_interval=1):
+        url = "http://%(host)s:%(port)s/" % {
+            "host": self.manager.services[1].host,
+            "port": self.manager.services[1].port
+        }
+        while retry_count:
+            response_status = None
+            retry_count = retry_count - 1
+            try:
+                response = urllib.request.urlopen(url)
+                response_status = response.getcode()
+            except urllib.error.HTTPError as error:
+                response_status = error.code
+            except urllib.error.URLError:
+                pass
+
+            if response_status == status:
+                return True
+            time.sleep(retry_interval)
+
+        return False
+
+    def _test_password(self, password, expected):
+        # Set the password in the Password Server
+        response = self._update_password(password)
+        self.assertEqual(200, response)
+
+        # Reboot the instance
+        self.manager.reboot_instance()
+
+        # Check if the password was set properly
+        response = self._run_remote_command(expected, 'echo 1')
+        self.assertEqual('1', response.strip())
+
+    @test_util.requires_service('cloudstack')
+    def test_update_password(self):
+        # Get the password from the metadata
+        password = self.manager.get_metadata()['admin_pass']
+
+        with self.manager.instantiate_mock_services():
+            # Wait until the web service starts serving requests
+            self.assertTrue(self._wait_for_service_status(400))
+
+            # Set a new password in Password Server and test if the
+            # plugin updates the password.
+            new_password = self._random_password()
+            self._test_password(password=new_password, expected=new_password)
+
+            # Remove the password from Password Server in order to check
+            # if the plugin keeps the last password.
+            self._test_password(password=None, expected=new_password)
+
+            # Restore the password in order to avoid problems with other
```

Comment:
This comment is superfluous, since the service is started for each separated scenario, so it will not have a persistent state across them.